### ISSUE TITLE
swarm leave hangs infinitely

### ISSUE THREAD

**Issue by kostrzewa9ld** [moby/moby#38824]
I have a 3-node cluster with around 15 services. When I run `docker swarm leave -f` (after docker  restart) on all the nodes in parallel then 1 out of 2 times the command hangs infinitely on one of the nodes. The command succeeds on the other two.

**Steps to reproduce the issue:**
1. Setup 3-node Swarm
2. Re-start docker daemon (`systemctl restart docker`)
3. Run `docker swarm leave -f` on all the nodes in parallel

**Describe the results you received:**
The command succeeds on two of the three nodes. On the third node tough the command first fails and then, when repeated, hangs infinitely.

**Describe the results you expected:**
The command should successfully remove all the nodes from the swarm.

**Additional information you deem important (e.g. issue happens only occasionally):**
It might be important that the leave command is executed right after restarting docker daemon (via systemd).

```
goroutine 1419 [chan receive, 47 minutes]:
github.com/docker/docker/vendor/github.com/docker/libnetwork.(*controller).AgentStopWait(0xc42011a400)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/vendor/github.com/docker/libnetwork/controller.go:383 +0x80
github.com/docker/docker/daemon.(*Daemon).DaemonLeavesCluster(0xc420162000)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/daemon/daemon.go:544 +0x85
github.com/docker/docker/daemon/cluster.(*Cluster).Leave(0xc420c400f0, 0x7f785b67cf01, 0x0, 0x0)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/daemon/cluster/swarm.go:409 +0x387
github.com/docker/docker/api/server/router/swarm.(*swarmRouter).leaveCluster(0xc421fd6de0, 0x7f78589db628, 0xc42236a240, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300, 0xc42236a180, 0x7f785b67cf11, 0x5)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/api/server/router/swarm/cluster_routes.go:48 +0x99
github.com/docker/docker/api/server/router/swarm.(*swarmRouter).(github.com/docker/docker/api/server/router/swarm.leaveCluster)-fm(0x7f78589db628, 0xc42236a240, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300, 0xc42236a180, 0x7f78589db628, 0xc42236a240)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/api/server/router/swarm/cluster.go:29 +0x6b
github.com/docker/docker/api/server/middleware.ExperimentalMiddleware.WrapHandler.func1(0x7f78589db628, 0xc42236a240, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300, 0xc42236a180, 0x7f78589db628, 0xc42236a240)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/api/server/middleware/experimental.go:27 +0xda
github.com/docker/docker/api/server/middleware.VersionMiddleware.WrapHandler.func1(0x7f78589db628, 0xc42236a1e0, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300, 0xc42236a180, 0xc42046d8a8, 0x7f785a1eeac3)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/api/server/middleware/version.go:62 +0x425
github.com/docker/docker/pkg/authorization.(*Middleware).WrapHandler.func1(0x7f78589db628, 0xc42236a1e0, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300, 0xc42236a180, 0xc42046da08, 0x7f785a1de2f7)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/pkg/authorization/middleware.go:59 +0x83d
github.com/docker/docker/api/server/middleware.DebugRequestMiddleware.func1(0x7f78589db628, 0xc42236a1e0, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300, 0xc42236a180, 0x7f78589db628, 0xc42236a1e0)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/api/server/middleware/debug.go:25 +0x969
github.com/docker/docker/api/server.(*Server).makeHTTPHandler.func1(0x7f785cc3e480, 0xc421d0c700, 0xc420c10300)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/api/server/server.go:137 +0x1c8
net/http.HandlerFunc.ServeHTTP(0xc4219b9be0, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300)
        /usr/local/go/src/net/http/server.go:1918 +0x46
github.com/docker/docker/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc421d965f0, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/vendor/github.com/gorilla/mux/mux.go:103 +0x22d
github.com/docker/docker/api/server.(*routerSwapper).ServeHTTP(0xc42216b2d0, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300)
        /root/rpmbuild/BUILD/src/engine/.gopath/src/github.com/docker/docker/api/server/router_swapper.go:29 +0x72
net/http.serverHandler.ServeHTTP(0xc420326680, 0x7f785cc3e480, 0xc421d0c700, 0xc420c10300)
        /usr/local/go/src/net/http/server.go:2619 +0xb6
net/http.(*conn).serve(0xc421c86460, 0x7f785cc40140, 0xc421d43a80)
        /usr/local/go/src/net/http/server.go:1801 +0x71f
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2720 +0x28a
```
Full [stack trace](https://raw.githubusercontent.com/kostrzewa9ld/issues/master/moby/38824/goroutine-stacks-2019-03-04T095838Z.log)
Maybe related to https://github.com/docker/for-linux/issues/499

**Output of `docker version`:**

```
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:20:16 2018
 OS/Arch:      linux/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:23:58 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

**Output of `docker info`:**

```
Containers: 9
 Running: 0
 Paused: 0
 Stopped: 9
Images: 12
Server Version: 18.03.1-ce
Storage Driver: devicemapper
 Pool Name: vg00-docker--pool
 Pool Blocksize: 524.3kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Udev Sync Supported: true
 Data Space Used: 5.646GB
 Data Space Total: 21.47GB
 Data Space Available: 15.83GB
 Metadata Space Used: 1.585MB
 Metadata Space Total: 218.1MB
 Metadata Space Available: 216.5MB
 Thin Pool Minimum Free Space: 2.147GB
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-327.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.642GiB
Name: node-10-9-4-116
ID: YW5M:LX7M:6NKZ:NN5F:EFX4:BOCK:VJIH:NWXB:4STF:LTVU:2DG5:HS46
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 30
 Goroutines: 62
 System Time: 2019-03-04T10:00:06.524008399Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 XXX:5000
 XXX:5002
 127.0.0.0/8
Registry Mirrors:
 http://XXX:5001/
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
CentOS 7.2 VM on KVM

[tail from syslog for dockerd](https://raw.githubusercontent.com/kostrzewa9ld/issues/master/moby/38824/syslog-dockerd)

**Comment by thaJeztah** [comment]
Let me close this ticket for now, as it looks like it went stale.
